### PR TITLE
feat: remove binary::Segment

### PR DESCRIPTION
**Pull request by haraldh**
Do not make another memory copy of the executables
before copying it in the final destination.

Signed-off-by: Harald Hoyer <harald@hoyer.xyz>

### CODE REVIEW

File: src/binary/component.rs [hunk 1]
```diff
@@ -96,32 +80,40 @@ impl Component {
...
 
-        Ok(Self {
-            entry: elf.entry as _,
-            segments,
-            pie,
-        })
+        Ok(component)
+    }
+
+    #[allow(dead_code)]
```
Comment (npmccallum) [inline src/binary/component.rs:96]: Why?

File: src/binary/component.rs [hunk 2]
```diff
@@ -96,32 +80,40 @@ impl Component {
...
+    pub fn filter_header(&self, type_: u32) -> impl Iterator<Item = &ProgramHeader> {
+        self.elf
+            .program_headers
+            .iter()
+            .filter(move |ph| ph.p_type == type_)
+    }
+
+    #[allow(dead_code)]
```
Comment (npmccallum) [inline src/binary/component.rs:104]: Why?

File: src/binary/component.rs (hunk 1 above)
Comment (haraldh) [reply src/binary/component.rs]: removed

File: src/binary/component.rs (hunk 2 above)
Comment (haraldh) [reply src/binary/component.rs]: removed

**Comment by haraldh** [comment]
Based on https://github.com/enarx/enarx-keepldr/pull/445

**Review by npmccallum — APPROVED** [review]

**Pull request merged**